### PR TITLE
databend 0.6.36-nightly (new formula)

### PR DESCRIPTION
**Pull request by dz85**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/databend.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+class Databend < Formula
+  desc "Elastic, reliable serverless data warehouse, offers blazing fast query"
+  homepage "https://databend.rs/"
+  url "https://github.com/datafuselabs/databend/archive/refs/tags/v0.6.36-nightly.tar.gz"
```
Comment (carlocab) [inline Formula/databend.rb:4]: Is there a stable version? We don't really package nightlies, along with the spirit of: https://docs.brew.sh/Acceptable-Formulae#stable-versions

File: Formula/databend.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  sha256 "73c4957bd51745b4bacd16f6b7324e14f74b68b1ff53b165ad780e8b23667876"
+  license "Apache-2.0"
+
+  livecheck do
+    url :url
+  end
+
+  depends_on "rustup-init" => :build
```
Comment (carlocab) [inline Formula/databend.rb:12]: This should depend on `rust`. We do have a handful of formulae that use `rustup`, but those are exceptions since they were needed by existing formulae. New formulae shouldn't need `rustup`.

File: Formula/databend.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    bins=%w[databend-query databend-benchmark databend-meta]
+    # https://github.com/rust-lang/cargo/issues/7124
+    system "#{ENV["CARGO_HOME"]}/bin/cargo", "build", "--release", *bins.map { |b| "--bin=#{b}" }, "--locked"
+    bins.each do |b|
+      bin.install "target/release/#{b}"
+    end
+  end
+
+  test do
+    system "true"
```
Comment (carlocab) [inline Formula/databend.rb:26]: This needs a proper test.

File: Formula/databend.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
+class Databend < Formula
+  desc "Elastic, reliable serverless data warehouse, offers blazing fast query"
+  homepage "https://databend.rs/"
+  url "https://github.com/datafuselabs/databend/archive/refs/tags/v0.6.36-nightly.tar.gz"
+  sha256 "73c4957bd51745b4bacd16f6b7324e14f74b68b1ff53b165ad780e8b23667876"
+  license "Apache-2.0"
+
+  livecheck do
+    url :url
+  end
```
Comment (carlocab) [inline Formula/databend.rb:10]: Why is this needed?

**Review by carlocab — COMMENTED** [review]
Thanks for the PR, @dz85. A few comments.

Note that our preferred commit message style here is
```
databend 0.6.36-nightly (new formula)
```

File: Formula/databend.rb (hunk 1 above)
Comment (dz85) [reply Formula/databend.rb:4]: no they are all nightly versions.

File: Formula/databend.rb (hunk 2 above)
Comment (dz85) [reply Formula/databend.rb:12]: but we need nightly version for rust which rustup can deal with well.

File: Formula/databend.rb (hunk 1 above)
Comment (carlocab) [reply Formula/databend.rb:4]: Then I don't think we can package this, sorry.

File: Formula/databend.rb (hunk 4 above)
Comment (dz85) [reply Formula/databend.rb:10]: to auto update

File: Formula/databend.rb (hunk 2 above)
Comment (carlocab) [reply Formula/databend.rb:12]: That's another reason that makes this unsuitable for packaging in Homebrew.

**Comment by carlocab** [comment]
Sorry, but we are passing on this for reasons discussed above. Thanks for the PR anyway.

If you wish, you can host this formula in your own tap. Here are some docs to help you get started:

https://docs.brew.sh/Taps
https://docs.brew.sh/Interesting-Taps-and-Forks
https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap
https://brew.sh/2020/11/18/homebrew-tap-with-bottles-uploaded-to-github-releases/

File: Formula/databend.rb (hunk 4 above)
Comment (carlocab) [reply Formula/databend.rb:10]: I think you misunderstood me: my point was the the livecheck should already work even if you delete these three lines.

File: Formula/databend.rb (hunk 2 above)
Comment (dz85) [reply Formula/databend.rb:12]: Ok I get it, thx for nice explaining.

File: Formula/databend.rb (hunk 4 above)
Comment (carlocab) [reply Formula/databend.rb:10]: Yes, livecheck checks `url` by default, so there is no need to add this.
